### ISSUE TITLE
MCMC diagnostic plots

### ISSUE THREAD

**Issue by zenith378** [bat/BAT.jl#432]
Hi,

I am reaching out to inquire about the possibility of incorporating methods for plotting diagnostic figures related to MCMC tuning in this project (e.g. trace and acf plots).

While exploring the project, I came across an example from four years ago that seems to demonstrate a MCMCDiagnostics method ([Example Link](https://github.com/bat/BAT.jl/blob/63553c38f9069a0a9fe00541b642b20163fcc0a3/examples/dev-internal/plotting_examples.jl#L215C1-L216C1)). However, I could not find this method in the documentation, and my attempts to utilize it based on the example were unsuccessful.

Could it be possible to have documentation or an updated example on the MCMCDiagnostics method? Alternatively, if this feature is not present, would it be possible to consider adding methods for generating trace and acf plots during MCMC tuning?

Thanks in advance.

**Comment by oschulz** [comment]
Yes, we definitely want to add better MCMC diagnostic plots, it's been on our mind for a while, and you're right, we should really get on this. :-)

**Comment by penelopeysm** [comment]
@oschulz If you're still interested in this several years on :-) I would suggest that the easiest way is to convert `DensitySampleVector` into a different object that provides this functionality. Traditionally this would probably be MCMCChains.jl, but that has various drawbacks of its own and the more modern replacement is [FlexiChains.jl](https://github.com/penelopeysm/FlexiChains.jl). I think a pretty small extension would be a quick win, and can be implemented either on your side or mine (depends on who wants to be responsible for the docs, I'm very happy to host it though).

**Comment by oschulz** [comment]
HI @penelopeysm - yes, I am, and sorry for the long long delay on this. We are in fact getting close to merge-ready on a new Makie-based plotting backend for BAT that will be able to do things like this, also online, during sampling.

I'm also very happy to support FlexiChains.jl via a package extension. Would you like to make a PR? Otherwise I can also put it on the list of our current updates - I'm currently doing a full AI-assisted review/fix/improve campaign across BAT and related packages, also well advanced (some things in other packages need to be polished before I can finalize the BAT changes). I can make FlexiChains support part of it.

**Comment by penelopeysm** [comment]
> sorry for the long long delay on this

No apology needed! I actually have no skin in the game at all, as I'm not a practitioner. I used to work on Turing, and FlexiChains was made with that in mind, but there's really no reason it should only be restricted to Turing, so I've just been on a hunt for other packages to integrate with.

I could try to put a PR together some time this week :)
